File: binary_search.c
#include <stdio.h>

// Note: Index starts from 0. 

void binary_search(int arr[], int to_find, int low, int high)
{
	low = 0;
	int mid_point = (low+high)/2;

	while(1){
		if(high < low){
			printf("Not Found\n");
			break;
		}
		if (arr[mid_point] == to_find){
			printf("Found at Index %d\n",mid_point);
			break;
		}
		else if (to_find < arr[mid_point]){
			high = mid_point -1;
			mid_point = (low+high) / 2;

		}
		else {
			low = mid_point + 1;
			mid_point = (low+high)/2;
		}

	}

}

int main(){

	int arr[] = {1,2,4,6,7,8};

	//Basic Tests below. 
	
	binary_search(arr,6,0,5); // will print "Found at index 3"
	binary_search(arr,21,0,5); // will print "Not Found"
	return 0;
}

File: binary_search.py

#Binary Search Algorithm implemented in python. It is implemented as a function. To run it pass to_find and input_list arguments to the function.
#The function will return the index of where the value is found. Else will return "Not Found" if the value is not in the list.

def binarySearch(to_find,input_list,low=None,high=None):
	if not low:
		low = 0
	if not high:
		high = len(input_list) - 1

	mid_point = (low+high) // 2

	while True:

		if high < low:
			return "Not Found"
			break

		if input_list[mid_point] == to_find:
			# print('Found at Index ' + str(mid_point))
			return mid_point
			break
		elif to_find < input_list[mid_point]:
			high =  mid_point - 1
			mid_point = (low + high) // 2
		else:
			low = mid_point + 1
			mid_point = (low + high) // 2

	





